scripts/generate_exp.py

```diff
@@ -0,0 +1,65 @@
+"""
+ Copyright (C) 2023 Intel Corporation
+
+ Part of the Unified-Runtime Project, under the Apache License v2.0 with LLVM Exceptions.
+ See LICENSE.TXT
+ SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+"""
+import sys
+from util import makoWrite
+import re
+
+usage = """
+OVERVIEW: A tool to aid in adding a new experimental feature to the Unified Runtime Specification.
+
+USAGE: python generate_exp.py [options] <name>
+
+    <name> - must be lowercase and kebab case i.e. command-buffer
+
+OPTIONS:
+    --help, -h              - Display usage and valid options available
+"""
+
+if __name__ == "__main__":
+    if len(sys.argv) < 2:
+        print("No name provided for the experimental feature.")
+        print(usage)
+        sys.exit(1)
+    
+    if len(sys.argv) > 2:
+        print("Too many arguments.")
+        print(usage)
+        sys.exit(1)
+
+    if "-h" in sys.argv or "--help" in sys.argv:
+        print(usage)
+        sys.exit(0)
+
+    exp_feat_name = sys.argv[1]
```

Comment:
Always use [argparse](https://docs.python.org/3/library/argparse.html) to handle command-line arguments in Python.

```suggestion
    parser = argparse.ArgumentParser(description="A tool to aid in adding a new experimental feature to the Unified Runtime Specification.")
    parser.add_argument('name', help="name of the new experinmental feature")
    args = parser.parse_args()
    exp_feat_name = args.name. # although just using args.name is preferable
```

does pretty much same thing while automatically generating the usage help message

Reply:
You're right - my original thinking was it would be simpler to not use it, but no point reinventing the wheel.